lib/Launcher.js

```diff
@@ -373,6 +390,48 @@ function waitForWSEndpoint(chromeProcess, timeout, preferredRevision) {
   });
 }
 
+/**
+ * @param {string} browserUrl
+ * @return {!Promise<string>}
+ */
+function getWSEndpoint(browserUrl) {
+  let resolve, reject;
+  const endpointUrl = URL.resolve(browserUrl, '/json/version');
+  const requestOptions = Object.assign(URL.parse(endpointUrl), { method: 'GET' });
+  const promise = new Promise((res, rej) => { resolve = res; reject = rej; });
+
+  function onGetWsEndpointDone(data) {
+    try {
+      const {webSocketDebuggerUrl} = JSON.parse(data);
+      resolve(webSocketDebuggerUrl);
+    } catch (e) {
+      onGetWsEndpointError(e);
```

Comment:
I don't think we need special error forwarding here.

Reply:
This way we have the same user friendly error message also is case of JSON parsing error failure. Example

```
Failed to fetch browser webSocket url from http://127.0.0.1:21222/: SyntaxError: Unexpected token < in JSON at position 0
  Stack:
Error: Failed to fetch browser webSocket url from http://127.0.0.1:21222/: SyntaxError: Unexpected token < in JSON at position 0
    at onGetWsEndpointError (/Users/lf/Projects/puppeteer/lib/Launcher.js:413:12)
    at onGetWsEndpointDone (/Users/lf/Projects/puppeteer/lib/Launcher.js:408:7)
    at IncomingMessage.res.on (/Users/lf/Projects/puppeteer/lib/Launcher.js:426:25)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:139:11)
    at process._tickCallback (internal/process/next_tick.js:181:9)
```